packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -576,6 +583,7 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
     const { subnetIds, hasPublic } = props.vpc.selectSubnets(props.vpcSubnets);
     const asgProps: CfnAutoScalingGroupProps = {
+      autoScalingGroupName: props.autoScalingGroupName,
```

Comment:
Should this be passed to the base class as the physical name and such?

Reply:
Yes you are right, I missed that the autoScalingGroupName property is a the resource physical name since the  [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html#cfn-autoscaling-autoscalinggroup-autoscalinggroupname) does not note the usual disclaimer: "If you specify a name, you can't perform updates that require replacement of this resource." . I have deployed a named autoscaling group and verified the name is the physical id.